tensorboard/plugins/image/summary_v2.py

```diff
@@ -113,18 +113,22 @@ def lazy_tensor():
             tf.debugging.assert_non_negative(max_outputs)
             images = tf.image.convert_image_dtype(data, tf.uint8, saturate=True)
             limited_images = images[:max_outputs]
-            encoded_images = tf.map_fn(
-                tf.image.encode_png,
-                limited_images,
-                dtype=tf.string,
-                name="encode_each_image",
-            )
-            # Workaround for map_fn returning float dtype for an empty elems input.
-            encoded_images = tf.cond(
-                tf.shape(input=encoded_images)[0] > 0,
-                lambda: encoded_images,
-                lambda: tf.constant([], tf.string),
-            )
+            if tf.compat.forward_compatible(2023, 5, 1):
```

Comment:
The fixed time here felt a bit absurd. Is this serving one time purpose? Will this line be removed later?

Reply:
Yes it will be removed later.

In general TF provides a 3 week window for forward compatibility, per https://www.tensorflow.org/api_docs/python/tf/compat/forward_compatible
The branch ensures that the HEAD of python produces GraphDefs that an earlier version of serving infra can still execute.
It doesn't check against the date of the system, but there is a date stamp baked into TensorFlow source code that is updated daily.

We remove these forward compatible branches (long) after the window expires. That's why they don't show up frequently in the code search. Here are a few left overs:

https://github.com/tensorflow/tensorflow/search?q=forward_compatible

Reply:
Thanks for the code pointers. To confirm, with the new changes of encode_png, are you claiming that we don't need map_fn and the workaround afterward? 
(The line will be removed but with the "else" block removed not line 117)

Reply:
Could you add a TODO(b/276803093) comment here? I'm not sure how difficult it would be - is it possible to also update the unit test for this change?

For presubmit testing, we can send you a command via chat, thanks!

Reply:
Done. I added a new unit test. There didn't appear to be coverage for vectorized input of batch dim size > 1.